dask/dataframe/core.py

```diff
@@ -3107,7 +3140,49 @@ def elemwise(op, *args, **kwargs):
         with raise_on_meta_error(funcname(op)):
             meta = partial_by_order(*parts, function=op, other=other)
 
-    return new_dd_object(dsk, _name, meta, divisions)
+    result = new_dd_object(dsk, _name, meta, divisions)
+    return handle_out(out, result)
+
+
+def handle_out(out, result):
+    """ Handle out parameters
+
+    If out is a dask.DataFrame, dask.Series or dask.Scalar then
+    this overwrites the contents of it with
+    the result
+    """
+    if isinstance(out, tuple):
+        if len(out) == 1:
+            out = out[0]
+        elif len(out) > 1:
+            raise NotImplementedError("The out parameter is not fully supported")
+        else:
+            out = None
+
+    if out is not None and type(out) != type(result):
+        raise NotImplementedError(
+            "Mismatched types between result and out parameter. "
+            "out=%s, result=%s" % (str(type(out)), str(type(result))))
+
+    if isinstance(out, DataFrame):
+        if len(out.columns) != len(result.columns):
+            raise NotImplementedError(
```

Comment:
`ValueError`?

Reply:
I think that this should probably raise a `ValueError` in both cases.  My apologies for the earlier implementation.